sdks/python/apache_beam/runners/interactive/pipeline_instrument.py

```diff
@@ -95,17 +120,131 @@ def instrumented_pipeline_proto(self):
     """Always returns a new instance of portable instrumented proto."""
     return self._pipeline.to_runner_api(use_fake_coders=True)
 
+  def _required_components(self, pipeline_proto, required_transforms_ids):
```

Comment:
Hi Sam, this could be achieved by building a pipeline fragment using `PipelineFragment([all_unbounded_source_output_pcolls])`.
The `pipeline_fragment` module is included in PR: https://github.com/apache/beam/pull/10201.

Reply:
ack

Reply:
I can change it after your PR goes through